WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,180 @@
+                    success: SuccessResponseBlock,
+                    failure: FailureReponseBlock) -> NSProgress?
+    {
+        let progress = NSProgress()
+        progress.totalUnitCount = 1
+        let xmlRPCParameters = XMLRPCCallParameters(method: method, parameters: parameters)
+        let task = sessionManager.POST(apiBaseURLString, parameters: xmlRPCParameters, success: { (dataTask, result) in
+            success(responseObject: result, httpResponse: dataTask.response as? NSHTTPURLResponse)
+            progress.completedUnitCount = 1
+            }, failure: { (dataTask, error) in
+                failure(error: error, httpResponse: dataTask?.response as? NSHTTPURLResponse)
+                progress.completedUnitCount = 1
+            }
+        )
+        if let task = task {
+            progress.cancellationHandler = {
+                task.cancel()
+            }
+            return progress
+        } else {
+            return nil
+        }
+    }
+}
+
+private final class XMLRPCCallParameters {
+    let method: String
+    let parameters: [AnyObject]?
+    let streamingCacheFile: NSURL?
+
+    init(method:String, parameters:[AnyObject]? = nil, streamingCacheFile: NSURL? = nil) {
+        self.method = method
+        self.parameters = parameters
+        self.streamingCacheFile = streamingCacheFile
+    }
+}
+
+/**
+ Error constants for the WordPress XMLRPC API
+
+ - CallNeedtoBePOST:                   The parameters sent to the server where invalid
+ - RequestSerializationFailed:     The serialization of the request failed
+ - Unknown:                        Unknow error happen
+ */
+@objc public enum WordPressOrgXMLRPCApiError: Int, ErrorType {
+    case CallNeedsToBePOST
+    case InvalidArguments
+    case RequestSerializationFailed
+    case Unknown
+}
+
+/// A custom serializer to handle XMLRPC responses
+final class WordPressOrgXMLRPCResponseSerializer: AFHTTPResponseSerializer
+{
+    override init() {
+        super.init()
+        self.acceptableContentTypes = ["application/xml", "text/xml"]
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override func responseObjectForResponse(response: NSURLResponse?, data: NSData?, error: NSErrorPointer) -> AnyObject? {
+        let responseObject = super.responseObjectForResponse(response, data: data, error: error)
+        guard error.memory == nil,
+            let unwrappedData = data else {
+            return responseObject
+        }
+        let decoder = WPXMLRPCDecoder(data:unwrappedData)
+
+        guard !decoder.isFault(),
+            let responseXML = decoder.object() else {
+            let decoderError = decoder.error()
+            error.memory = decoderError
+            return responseObject
+        }
+
+        return responseXML
+    }
+}
+
+/// A custom serializer to handle XMLRPC requests
+final class WordPressOrgXMLRPCRequestSerializer: AFHTTPRequestSerializer
```

Comment:
I keep going back and forth on this but I'm wondering if the custom serializer complicates things too much in this case. It requires us to have that extra `XMLRPCCallParameters` and do a lot of manual checking, I guess because XML-RPC is a bit different from what the serializers were designed for.

A simpler alternative would be to encode the request directly and call `dataTaskWithRequest`. (Note I haven't tested this code)

https://gist.github.com/koke/6aa615a3eb4b166d677170e4b6ecca1d